sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/HttpInstrumentationOptions.java

```diff
@@ -0,0 +1,303 @@
+ *     <li>Response status code</li>
+ *     <li>Error details if the request fails</li>
+ *     <li>Time it takes to receive response</li>
+ *     <li>Correlation identifiers</li>
+ * </ul>
+ *
+ The following information is recorded on detailed HTTP logs:
+ * <ul>
+ *     <li>Request method, URI, and body size. URI is sanitized based on allowed query parameters configurable with {@link #setAllowedQueryParamNames(Set)} and {@link #addAllowedQueryParamName(String)}</li>
+ *     <li>Response status code and body size</li>
+ *     <li>Request and response headers from allow-list configured via {@link #setAllowedHeaderNames(Set)} and {@link #addAllowedHeaderName(HttpHeaderName)}.</li>
+ *     <li>Error details if the request fails</li>
+ *     <li>Time it takes to receive response</li>
+ *     <li>Correlation identifiers</li>
+ *     <li>When content logging is enabled via {@link #setContentLoggingEnabled(boolean)}: request and response body, and time-to-last-byte</li>
+ * </ul>
+ *
+ * Client libraries auto-discover global OpenTelemetry SDK instance configured by the java agent or
+ * in the application code. Just create a client instance as usual as shown in the following code snippet:
+ *
+ * <p><strong>Clients auto-discover global OpenTelemetry</strong></p>
+ *
+ * <!-- src_embed io.clientcore.core.telemetry.useglobalopentelemetry -->
+ * <pre>
+ *
+ * AutoConfiguredOpenTelemetrySdk.initialize&#40;&#41;;
+ *
+ * SampleClient client = new SampleClientBuilder&#40;&#41;.build&#40;&#41;;
+ *
+ * &#47;&#47; this call will be traced using OpenTelemetry SDK initialized globally
+ * client.clientCall&#40;&#41;;
+ *
+ * </pre>
+ * <!-- end io.clientcore.core.telemetry.useglobalopentelemetry -->
+ * <p>
+ *
+ * Alternatively, application developers can pass OpenTelemetry SDK instance explicitly to the client libraries.
+ *
+ * <p><strong>Pass configured OpenTelemetry instance explicitly</strong></p>
+ *
+ * <!-- src_embed io.clientcore.core.telemetry.useexplicitopentelemetry -->
+ * <pre>
+ *
+ * OpenTelemetry openTelemetry = AutoConfiguredOpenTelemetrySdk.initialize&#40;&#41;.getOpenTelemetrySdk&#40;&#41;;
+ * HttpInstrumentationOptions&lt;OpenTelemetry&gt; instrumentationOptions = new HttpInstrumentationOptions&lt;OpenTelemetry&gt;&#40;&#41;
+ *     .setProvider&#40;openTelemetry&#41;;
+ *
+ * SampleClient client = new SampleClientBuilder&#40;&#41;.instrumentationOptions&#40;instrumentationOptions&#41;.build&#40;&#41;;
+ *
+ * &#47;&#47; this call will be traced using OpenTelemetry SDK provided explicitly
+ * client.clientCall&#40;&#41;;
+ *
+ * </pre>
+ * <!-- end io.clientcore.core.telemetry.useexplicitopentelemetry -->
+ *
+ * @param <T> The type of the instrumentation provider. When using OpenTelemetry, this is {@code io.opentelemetry.api.OpenTelemetry}. If not provided
```

Comment:
Do we actually need this type to be generic or do we really just need a property in the class to use a bounded generic (`Class<? extends Provider>`)